### PR TITLE
Some refactoring and better typings

### PR DESCRIPTION
**Pull request by brunohpaiva**
### Code changes

* Better typing:
  * `src/rest/router.ts`
  * discord.js extensions
  * `Command` class
  * `CommandHandler` class
  * `lib/util/converters.ts`
* Simplified route manager and refactored all routes.
* Now fetching discord-akairo admin users id from `ADMINS` env variable.
* Added a `POSTGRES_USER` env variable and fallback to `"postgres"`.
* Moved discord-akairo type casters to separated files.
* Removed useless `SET uid=$2` statement in `updateBlacklist` method from `ClientUtils` class.
* Now using enums for Websocket event types and states.
* Refactored all commands.
* Prettier formatting.

### Misc

* Moved some dependencies to the `devDependecies` property in `package.json`.
* Added `typescript` and `ts-node` dependencies to always use the correct version across other contributors.
* Added `prettier` dependency & config file and also added a npm script `format`. The config file is `.prettierrc`.

### CODE REVIEW

**Comment by brunohpaiva** [comment]
Sk1er's guild module needs testing

**Review by GamingGeek — CHANGES_REQUESTED** [review]

File: lib/providers/postgres.ts [hunk 1]
```diff
@@ -13,11 +13,14 @@ export class PostgresProvider extends Provider {
...
   idColumn: string;
   dataColumn: string;
+
   constructor(
     db: Client,
     tableName: string,
     { idColumn = "gid", dataColumn = "data" } = {}
   ) {
+    4;
+    4;
```
[the remark this answers is not in the document]
Comment (brunohpaiva) [reply lib/providers/postgres.ts:23]: LMAO MY KEYBOARD

**Comment by brunohpaiva** [comment]
Added `POSTGRES_DB` and `OWNER` env variables, you will need to set them on your env files

**Comment by GamingGeek** [comment]
There's a few changes that need to be made but I'll just merge and fix them myself since there's not many left.

Thanks :)

**Pull request merged**